fem/fe_coll.cpp

```diff
@@ -1742,7 +1742,11 @@ L2_FECollection::L2_FECollection(const int p, const int dim, const int type,
    }
    OtherDofOrd = NULL;
 
-   if (dim == 1)
+   if (dim == 0)
+   {
+      L2_Elements[Geometry::POINT] = new PointFiniteElement;
```

Comment:
I don't really have enough context to know if this is sufficient to cover all the use cases of an L2 space.
It was sufficient to get rid of the segfault, and to assign one dof to my simple point mesh (see the PR description).

E.g.
* does it make sense to have a higher order L2 space on a point?
* are there other fields that need to be initialized here ( such as ``Tr_Elements``, which looks like it is only for lower dimensional faces)?